autoreadme.go

```diff
@@ -207,6 +217,53 @@ func getDoc(dir string) (Doc, error) {
 	}, nil
 }
 
+func renderExamples(bi *build.Package) (map[string]string, error) {
+	filter := func(fi os.FileInfo) bool {
+		if fi.IsDir() {
+			return false
+		}
+		nm := fi.Name()
+		for _, f := range bi.XTestGoFiles {
+			if nm == f {
+				return true
+			}
+		}
+		return false
+	}
+	pkgs, err := parser.ParseDir(token.NewFileSet(), bi.Dir, filter, parser.ParseComments)
+	if err != nil {
+		return nil, err
+	}
+	var files []*ast.File
+	for _, pkg := range pkgs {
+		for _, f := range pkg.Files {
+			files = append(files, f)
+		}
+	}
+
+	examples := map[string]string{}
+	for _, ex := range doc.Examples(files...) {
+		examples[ex.Name] = renderExample(ex)
```

Comment:
Let's go with a struct that just has a Name and Code field. Replace _ with " " in the Name field but leave it in the map key. That way it's easy to add the Doc field later without breaking anyone's templates.